pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -122,51 +115,30 @@ protected StreamPartitionMsgOffsetFactory getStreamPartitionMsgOffsetFactory(LLC
     return StreamConsumerFactoryProvider.create(streamConfig).createStreamMsgOffsetFactory();
   }
 
-  // We need to make sure that we never create multiple FSMs for the same segment
-  // Obtain locks based on segment name, so as to disallow same segment names entering together
```

Comment:
The locks were added in this PR: https://github.com/apache/pinot/pull/4159
At that time, fsmMap was also available. Am I missing a point here or could've we done the same thing in the PR#4159?

Reply:
Yes, we should use `ConcurrentHashMap.computeIfAbsent()` to achieve this instead of adding lock. The old code uses `get()` which doesn't provide the same guarantee